api/services/registration_service.py

```diff
@@ -78,21 +78,27 @@ def update_registration(self, registration, jsonData):
         """Updates the registration data
 
         Args:
-            registration (onject): Takes registration as object
+            registration (object): Takes registration as object
             jsonData (json): Takes request json for updating registration fields
         """
+        if not self.has_default_program_selection:
+            registration.program_id = self.selected_program_id
```

Comment:
Updating the `program_id` only if the user has selected any program, else there is not need to update the program id.